neps/nep-0452.md

```diff
@@ -0,0 +1,231 @@
+///
+/// Arguments:
+/// * `account_id` the account that should receive the linkdrop assets.
+///
+/// Returns `true` if the claim was successful meaning all assets were sent to the `account_id`.
+function claim(account_id: string) -> Promise<boolean>;
+
+/// Creates a new NEAR account and transfers all assets linked to the signer’s public key to
+/// the *newly created account*. If the transfer fails for whatever reason, it is up to the
+/// smart contract developer to choose what should happen. For example, the contract can
+/// choose to keep the assets or return them to the original linkdrop creator.
+///
+/// Requirements:
+/// * The predecessor account *MUST* be the current contract ID.
+/// * The assets being sent *MUST* be associated with the signer’s public key.
+/// * The assets *MUST* be sent to the `new_account_id` passed in.
+/// * The newly created account *MUST* have a new access key added to its account (either
+///   full or limited access) in the same receipt that the account was created in.
+/// * The Public key must be in a binary format with base58 string serialization with human-readable curve.
+///   The key types currently supported are secp256k1 and ed25519. Ed25519 public keys accepted are 32 bytes
+///   and secp256k1 keys are the uncompressed 64 format.
+///
+/// Arguments:
+/// * `new_account_id`: the valid NEAR account which is being created and should
+///   receive the linkdrop assets
+/// * `new_public_key`: the valid public key that should be used for the access key added to the newly created account (serialized with borsh).
+///
+/// Returns `true` if the claim was successful meaning the `new_account_id` was created and all assets were sent to it.
+function create_account_and_claim(new_account_id: string, new_public_key: string) -> Promise<boolean>;
+```
+
+## Reference Implementation
+
+Below are some references for linkdrop contracts.
+
+- [Link Drop Contract](https://github.com/near/near-linkdrop)
+
+- [Keypom Contract](https://github.com/keypom/keypom)
+
+## Security Implications
+
+1. Linkdrop Creation
+   Linkdrop creation involves creating keypairs that, when used, have access to assets such as $NEAR, FTs, NFTs, etc. These keys should be limited access and restricted to specific functionality. For example, they should only have permission to call `claim` and `create_account_and_claim`. Since the keys allow the holder to sign transactions on behalf of the linkdrop contract, without the proper security measures, they could be used in a malicious manner (for example executing private methods or owner-only functions).
+
+Another important security implication of linkdrop creation is to ensure that only one key is mapped to a set of assets at any given time. Externally, assets such as FTs, and NFTs belong to the overall linkdrop contract account rather than a specific access key. It is important to ensure that specific keys can only claim assets that they are mapped to.
```

Comment:
If you add indentation (3 or 4 spaces) to this paragraph then you will fix the following markdown lint errors

Reply:
@BenKurrek ☝️ finish those markdown lint errors please

Reply:
I will fix linting errors post consensus